scripts/prepare-subcharts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -e
+
+# Pull in the subcharts required by a chart and apply any patches on subcharts with fixed version numbers
+
+if [[ -z $1 ]]; then
+	echo "No directory provided to pull in subcharts"
+	exit 1
+fi
+
+f=$1
+
+if ! [[ -f ${f}/charts/requirements.yaml ]]; then
+	echo "Chart ${CHART} has no dependencies; skipping step to pull in dependencies"
+	exit 0
+fi
+
+# Apply the patch on the requirements.yaml
+for file in $(find ./${f} -type f -name "*.patch"); do
+	awk '/diff.*requirements.yaml/{flag=1;next}/diff/{flag=0}flag' $file >> ${f}/requirements.patch
```

Comment:
Totally agree with how this code is unreadable / hard to understand. This definitely highlights why we need to move to using Go scripts instead of using Bash scripts at some point in the near future.

For this release, I'm not sure if there's any better way to simplify / refactor this awk statement; do you think it would be easier to understand if I added a comment? i.e.
```suggestion
        # This awk statement extracts the contents between the first line that matches the regex `diff.*requirements.yaml`
        # and the first line that matches the regex `diff` after it. This corresponds to the contents of any patches on the requirements.yaml file.
        # For more information about awk flags, see https://www.thelinuxrain.com/articles/how-to-use-flags-in-awk
	awk '/diff.*requirements.yaml/{flag=1;next}/diff/{flag=0}flag' $file >> ${f}/requirements.patch
```

Reply:
Can you add just the first 2 comment lines.

Reply:
will be updated in next push